### PR TITLE
MacOS credential access and defense evasion

### PR DESCRIPTION
**Pull request by bfuzzy1**
- Adding keychain dump
- Adding disable security updates (MacOS)
- Adding MacOS credential prompt

### CODE REVIEW

**Review by privateducky — APPROVED** [review]
Solid. Deferring to @khyberspache for final merge.

File: credential-access/5c90420e-7972-4125-833b-16c89d5c303f.yml [hunk 1]
```diff
@@ -64,9 +64,15 @@ platforms:
   linux:
     sh:
       command: 'read -sp "Enter Password: " T1187 && echo $T1187 > T1187.txt'
+  darwin:
+    sh:
+      command: >-
```
Comment (khyberspache) [inline credential-access/5c90420e-7972-4125-833b-16c89d5c303f.yml:69]: We actually have the boiler plate for supporting osascript directly. We could probably add that in fairly easily, but it wouldn't know if we could write to file that way.

Something like this:
https://github.com/preludeorg/pneuma/pull/35

Thoughts?

File: defense-evasion/4c9a443f-796d-4b79-a452-9c68ec85188a.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+description: This procedure disables the installation of security updates.
+tactic: defense-evasion
+technique:
+  id: T1562.001
+  name: Disable or Modify Tools
+platforms:
+  darwin:
+    sh:
+      command: >-
+        defaults write /Library/Preferences/com.apple.SoftwareUpdate.plist
```
Comment (khyberspache) [inline defense-evasion/4c9a443f-796d-4b79-a452-9c68ec85188a.yml:12]: Do we need a semicolon at the end of this line, or even better an `&&`?

**Review by khyberspache — CHANGES_REQUESTED** [review]
A few questions but looks good overall

File: defense-evasion/4c9a443f-796d-4b79-a452-9c68ec85188a.yml (hunk 2 above)
Comment (bfuzzy1) [reply defense-evasion/4c9a443f-796d-4b79-a452-9c68ec85188a.yml:12]: This should run without issue as long as it's being ran from an elevated context. Not sure why it was parsed as two lines, but I just tested the same yaml in Operator with no issues on my end.

File: credential-access/5c90420e-7972-4125-833b-16c89d5c303f.yml (hunk 1 above)
Comment (bfuzzy1) [reply credential-access/5c90420e-7972-4125-833b-16c89d5c303f.yml:69]: Good point. I'm testing this out now on my end.

Comment (bfuzzy1) [reply credential-access/5c90420e-7972-4125-833b-16c89d5c303f.yml:69]: Removed bash version and added new osa executor on this one!

**Review by khyberspache — APPROVED** [review]

**Pull request merged**